### PR TITLE
OLS-261: Rename src/utils to rag

### PR DESCRIPTION
**Pull request by onmete**
## Description

The code in `token_handler.py` is not used anywhere at the moment. But it seems to be more RAG-related, hence the change.
The pleasant side-effect is we don't have spread utils in multiple places.

## Type of change

- [x] Refactor

### CODE REVIEW

**Review by tisnik — COMMENTED** [review]
it looks ok from my perspective

File: ols/rag/token_handler.py
Comment (bparees) [inline ols/rag/token_handler.py:1]: we're going to have other uses for the token handler beyond just counting tokens for RAG. We are also going to need to count tokens for conversation history and possibly other aspects of constraining the overall prompt length

Comment (onmete) [reply ols/rag/token_handler.py:1]: Ok, so I would move it to `ols/utils/token_handler.py`, WDYT?

Comment (bparees) [reply ols/rag/token_handler.py:1]: sounds good to me

**Review by tisnik — COMMENTED** [review]
ok

**Pull request merged**